src/classes/RD2_ValidationService.cls

```diff
@@ -575,4 +601,72 @@ public with sharing class RD2_ValidationService {
         return closeWonDateByRdId;
     }
 
+
+    /***
+    * @description Notification class containing validation errors 
+    */
+    public class Notification {
```

Comment:
Returning notifications is key to decoupling validation service from the trigger (W-8247722).